proto/pulumi/provider.proto

```diff
@@ -276,37 +293,74 @@ message CallResponse {
     map<string, ReturnDependencies> returnDependencies = 2;
 }
 
+// `CheckRequest` is the type of requests sent as part of [](pulumirpc.ResourceProvider.CheckConfig) and
+// [](pulumirpc.ResourceProvider.Check) calls. A `CheckRequest` primarily captures the URN and inputs of the resource
+// being checked. In the case of [](pulumirpc.ResourceProvider.CheckConfig), the URN will be the URN of the provider
+// resource being constructed, which may or may not be a [default provider](default-providers), and the inputs will be
+// the provider configuration.
 message CheckRequest {
-    string urn = 1;                  // the Pulumi URN for this resource.
-    google.protobuf.Struct olds = 2; // the old Pulumi inputs for this resource, if any.
+    // The URN of the resource whose inputs are being checked. In the case of
+    // [](pulumirpc.ResourceProvider.CheckConfig), this will be the URN of the provider resource being constructed,
+    // which may or may not be a [default provider](default-providers).
+    string urn = 1;
 
-    // the new Pulumi inputs for this resource.
-    //
-    // Note that if the user specifies the ignoreChanges resource option, the value of news passed
-    // to the provider here may differ from the values written in the program source. It will be pre-processed by
-    // replacing every ignoreChanges property by a matching value from the old inputs stored in the state.
+    // The old input properties or configuration for the resource, if any.
+    google.protobuf.Struct olds = 2;
+
+    // The new input properties or configuration for the resource, if any.
     //
-    // See also: https://www.pulumi.com/docs/concepts/options/ignorechanges/
+    // :::{note}
+    // If this resource has been specified with the
+    // [`ignoreChanges`](https://www.pulumi.com/docs/concepts/options/ignorechanges/), then the values in `news` may
```

Comment:
Should we link to the `ignoreChanges` section of the developer docs here instead? https://pulumi-developer-docs.readthedocs.io/latest/developer-docs/providers/implementers-guide.html#ignorechanges